sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala

```diff
@@ -367,6 +367,10 @@ class TungstenAggregationIterator(
     }
   }
 
+  // Updating average hashmap probe after processing input rows. So even the iterator of result
+  // rows is not consumed to the last record, we still can show the metric.
+  avgHashProbe.set(hashMap.getAverageProbesPerLookup())
```

Comment:
hmm, the peak memory of hashMap should not be changed during iterating the rows. But looks like the memory usage of the external sorter will change.  It might be meaningful to update `peakExecutionMemory` at the last record.

Maybe it is also good to use task completion listener to update it?